packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1036,28 +1055,30 @@ abstract class RenderSliver extends RenderObject {
           (!sizedByParent && debugDoingThisLayout))
         return true;
       assert(!debugDoingThisResize);
-      String contract, violation, hint;
+      DiagnosticsNode contract, violation, hint;
       if (debugDoingThisLayout) {
         assert(sizedByParent);
-        violation = 'It appears that the geometry setter was called from performLayout().';
-        hint = '';
+        violation = ErrorDescription('It appears that the geometry setter was called from performLayout().');
       } else {
-        violation = 'The geometry setter was called from outside layout (neither performResize() nor performLayout() were being run for this object).';
+        violation = ErrorDescription('The geometry setter was called from outside layout (neither performResize() nor performLayout() were being run for this object).');
         if (owner != null && owner.debugDoingLayout)
-          hint = 'Only the object itself can set its geometry. It is a contract violation for other objects to set it.';
+          hint = ErrorHint('Only the object itself can set its geometry. It is a contract violation for other objects to set it.');
```

Comment:
I'd consider this a description, since it's unclear what the user should try to resolve the error.

Reply:
@Hixie it looks like you wrote the line that originally gave this the variable name `hint`.
Do you think it should be called a hint with the current definition of what a hint is?

To make it clearer what action to take we could write something a bit redundant like.
`Look for where the geometry is being set outside outside the object refactor the logic to set the geometry from inside the object.`
For now set this as an ErrorDescription as
`hint = ErrorDescription(`
is a great breadcrumb to later go back and cleanup ErrorDescription objects.